main.go

```diff
@@ -112,16 +120,27 @@ func (l opSet) Set(s string) error {
 
 func main() {
 	flag.Parse()
+	encoder, ok := logLevelEncoders[*logLevelEncoder]
+	if !ok {
+		setupLog.Error(fmt.Errorf("invalid log level encoder: %v", *logLevelEncoder), "Invalid log level encoder")
+		os.Exit(1)
+	}
 
 	switch *logLevel {
 	case "DEBUG":
-		ctrl.SetLogger(crzap.New(crzap.UseDevMode(true)))
+		eCfg := zap.NewDevelopmentEncoderConfig()
+		eCfg.LevelKey = *logLevelKey
+		eCfg.EncodeLevel = encoder
+		ctrl.SetLogger(crzap.New(crzap.UseDevMode(true), crzap.Encoder(zapcore.NewConsoleEncoder(eCfg))))
 	case "WARNING", "ERROR":
-		setLoggerForProduction()
+		setLoggerForProduction(encoder)
 	case "INFO":
 		fallthrough
 	default:
-		ctrl.SetLogger(crzap.New(crzap.UseDevMode(false)))
+		eCfg := zap.NewProductionEncoderConfig()
+		eCfg.LevelKey = *logLevelKey
+		eCfg.EncodeLevel = encoder
+		ctrl.SetLogger(crzap.New(crzap.UseDevMode(false), crzap.Encoder(zapcore.NewJSONEncoder(eCfg)) ))
```

Comment:
I believe if encoder is set to `color` or `capitalcolor` option, output gets messed up with JSON encoder

Reply:
Output appears to be WAI, with the linux color encodings embedded in the JSON value: `"severity":"\u001b[34mINFO\u001b[0m"`

Reply:
I don't think this is intended: https://github.com/uber-go/zap/issues/611
Should we set encoding to console encoder if user specifies color?

Reply:
I think we can give users the room to decide what works for them. We aren't defaulting to color encoding, so this isn't an issue where we are configuring things out-of-the-box in a way that surprises users.

Given that the behavior is opt-in, I see no reason why we should assume there is no situation where this would be useful for a person. I could see us making things even more configurable in the future (e.g. specify whether users want console or JSON encoding). For now, though, log level is the most pressing issue, as it makes ingesting logs very difficult.